clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -376,67 +374,68 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
 
 class KernelBodyTransform : public TreeTransform<KernelBodyTransform> {
 public:
-  KernelBodyTransform(llvm::DenseMap<DeclaratorDecl *, DeclaratorDecl *> &Map,
+  KernelBodyTransform(std::pair<DeclaratorDecl *, DeclaratorDecl *> &MPair,
                       Sema &S)
-      : TreeTransform<KernelBodyTransform>(S), DMap(Map), SemaRef(S) {}
+      : TreeTransform<KernelBodyTransform>(S), MappingPair(MPair), SemaRef(S) {}
   bool AlwaysRebuild() { return true; }
 
   ExprResult TransformDeclRefExpr(DeclRefExpr *DRE) {
     auto Ref = dyn_cast<DeclaratorDecl>(DRE->getDecl());
-    if (Ref) {
-      auto NewDecl = DMap[Ref];
-      if (NewDecl) {
-        return DeclRefExpr::Create(
-            SemaRef.getASTContext(), DRE->getQualifierLoc(),
-            DRE->getTemplateKeywordLoc(), NewDecl, false, DRE->getNameInfo(),
-            NewDecl->getType(), DRE->getValueKind());
-      }
+    if (Ref && Ref == MappingPair.first) {
+      auto NewDecl = MappingPair.second;
+      return DeclRefExpr::Create(
+          SemaRef.getASTContext(), DRE->getQualifierLoc(),
+          DRE->getTemplateKeywordLoc(), NewDecl, false, DRE->getNameInfo(),
+          NewDecl->getType(), DRE->getValueKind());
     }
     return DRE;
   }
 
 private:
-  DeclMap DMap;
+  std::pair<DeclaratorDecl *, DeclaratorDecl *> MappingPair;
   Sema &SemaRef;
 };
 
-static FunctionDecl *CreateSYCLKernelFunction(ASTContext &Context,
-                                              StringRef Name,
-                                              ArrayRef<ParamDesc> ParamDescs) {
+static FunctionDecl *
+CreateOpenCLKernelDeclaration(ASTContext &Context, StringRef Name,
```

Comment:
Maybe it's worth commenting this function similarly to CreateOpenCLKernelBody while we are here?

Reply:
Hm, In comment for `CreateOpenCLKernelBody ` I described what we do inside the OpenCL kernel body but I don't see what I can write for `CreateOpenCLKernelDeclaration`. Do you have any suggestions?

Reply:
Sorry, In hindsight I think your correct the function name/definition itself is pretty self explanatory, so you can disregard my comment, apologies for wasting your time in this case.